### PR TITLE
[meshsync] GET MeshSync Resources Kinds

### PR DESCRIPTION
**Pull request by theBeginner86**
**Notes for Reviewers**

This PR adds new GET" `/api/system/meshsync/resources/kinds` get kinds of meshsync resources.
Would help in filtering support in UI.

Screenshot 2023-12-27 at 6 38 45 PM

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 032fcc7245da386117cbaac33e3f077bb428fb10 at: https://658c1968215c8c5b8d6c9f1d--meshery-docs-preview.netlify.app

File: server/handlers/meshsync_handler.go [hunk 1]
```diff
@@ -164,6 +164,117 @@ func (h *Handler) GetMeshSyncResources(rw http.ResponseWriter, r *http.Request,
...
+// ```?order = {field}``` orders on the passed field
+//
+// responses:
+// 200: []meshsyncResourcesKindsResponseWrapper
+
+func (h *Handler) GetMeshSyncResourcesKinds(rw http.ResponseWriter, r *http.Request, _ *models.Preference, _ *models.User, provider models.Provider) {
+	rw.Header().Set("Content-Type", "application/json")
+	enc := json.NewEncoder(rw)
+
+	limitstr := r.URL.Query().Get("pagesize")
```
Comment (MUzairS15) [inline server/handlers/meshsync_handler.go:188]: Please use `getPaginationParams` function defined in `server/handlers/utils.go`.

File: server/router/server.go [hunk 2]
```diff
@@ -280,8 +280,13 @@ func NewRouter(_ context.Context, h models.HandlerInterface, port int, g http.Ha
 
 	gMux.Handle("/api/system/meshsync/resources", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(h.GetMeshSyncResources), models.ProviderAuth))).
 		Methods("GET")
+<<<<<<< HEAD
```
Comment (MUzairS15) [inline server/router/server.go:283]: Conflicts not resolved.

File: server/models/handlers.go [hunk 3]
```diff
@@ -210,7 +210,11 @@ type HandlerInterface interface {
 	GetConnectionsOfEnvironmentHandler(w http.ResponseWriter, req *http.Request, prefObj *Preference, user *User, provider Provider)
 
 	GetMeshSyncResources(rw http.ResponseWriter, r *http.Request, prefObj *Preference, user *User, provider Provider)
+<<<<<<< HEAD
+	GetMeshSyncResourcesKinds(rw http.ResponseWriter, r *http.Request, prefObj *Preference, user *User, provider Provider)
+=======
 	DeleteMeshSyncResource(rw http.ResponseWriter, r *http.Request, prefObj *Preference, user *User, provider Provider)
+>>>>>>> upstream/master
```
Comment (MUzairS15) [inline server/models/handlers.go:217]: Same here.

File: server/models/meshsync.go [hunk 4]
```diff
@@ -10,3 +10,10 @@ type MeshSyncResourcesAPIResponse struct {
 	TotalCount int64                      `json:"total_count"`
 	Resources  []model.KubernetesResource `json:"resources"`
 }
+
+type MeshSyncResourcesKindsAPIResponse struct {
+	Kinds []string `json:"kinds"`
```
Comment (MUzairS15) [inline server/models/meshsync.go:15]: Please run `gofmt`

**Review by MUzairS15 — CHANGES_REQUESTED** [review]

File: server/handlers/meshsync_handler.go (hunk 1 above)
Comment (theBeginner86) [reply server/handlers/meshsync_handler.go]: Switched :+1

File: server/router/server.go (hunk 2 above)
Comment (theBeginner86) [reply server/router/server.go]: resolved.

**Comment by l5io** [comment]
🚀 Preview for commit c432590338bce301a9de90ab7ab14822ff7e0884 at: https://658c38be2473eb710c87edb3--meshery-docs-preview.netlify.app

File: server/handlers/meshsync_handler.go [hunk 5]
```diff
@@ -164,6 +164,91 @@ func (h *Handler) GetMeshSyncResources(rw http.ResponseWriter, r *http.Request,
...
+	}
+
+	response := &models.MeshSyncResourcesKindsAPIResponse{
+		Kinds:      kinds,
+		TotalCount: totalCount,
+		Page:       page,
+		PageSize:   limit,
+	}
+
+	rw.Header().Set("Content-Type", "application/json")
```
Comment (MUzairS15) [inline server/handlers/meshsync_handler.go:246]:
```suggestion
	rw.Header().Set("Content-Type", "application/json")
```
added twice.

**Review by MUzairS15 — APPROVED** [review]
Thanks.
LGTM

File: server/handlers/meshsync_handler.go (hunk 5 above)
Comment (MUzairS15) [reply server/handlers/meshsync_handler.go:246]:
```suggestion
```

**Pull request merged**